core/src/test/java/org/opensearch/sql/expression/datetime/TimeStampDiffTest.java

```diff
@@ -0,0 +1,364 @@
+
+        //Test for all interval options
+        Arguments.of(
+            "MICROSECOND",
+            new ExprDatetimeValue("2000-01-06 00:00:00"),
+            new ExprDatetimeValue("2000-01-06 00:00:00.000123"),
+            123),
+        Arguments.of(
+            "SECOND",
+            new ExprDatetimeValue("2000-01-06 00:00:00"),
+            new ExprDatetimeValue("2000-01-06 00:00:12"),
+            12),
+        Arguments.of(
+            "MINUTE",
+            new ExprDatetimeValue("2000-01-06 00:00:00"),
+            new ExprDatetimeValue("2000-01-06 00:11:12"),
+            11),
+        Arguments.of(
+            "HOUR",
+            new ExprDatetimeValue("2000-01-06 00:00:00"),
+            new ExprDatetimeValue("2000-01-06 10:11:12"),
+            10),
+        Arguments.of(
+            "DAY",
+            new ExprDatetimeValue("2000-01-01 00:00:00"),
+            new ExprDatetimeValue("2000-01-06 10:11:12"),
+            5),
+        Arguments.of(
+            "WEEK",
+            new ExprDatetimeValue("2000-01-01 00:00:00"),
+            new ExprDatetimeValue("2000-01-31 10:11:12"),
+            4),
+        Arguments.of(
+            "MONTH",
+            new ExprDatetimeValue("2000-01-01 00:00:00"),
+            new ExprDatetimeValue("2000-12-31 10:11:12"),
+            11),
+        Arguments.of(
+            "QUARTER",
+            new ExprDatetimeValue("2000-01-01 00:00:00"),
+            new ExprDatetimeValue("2000-12-31 10:11:12"),
+            3),
+        Arguments.of(
+            "YEAR",
+            new ExprDatetimeValue("1999-01-01 00:00:00"),
+            new ExprDatetimeValue("2000-12-31 10:11:12"),
+            1),
+
+        //Test around Leap Year
+        Arguments.of(
+            "DAY",
+            new ExprDatetimeValue("2019-02-28 00:00:00"),
+            new ExprDatetimeValue("2019-03-01 00:00:00"),
+            1),
+        Arguments.of(
+            "DAY",
+            new ExprDatetimeValue("2020-02-28 00:00:00"),
+            new ExprDatetimeValue("2020-03-01 00:00:00"),
+            2),
+
+        //Test around year change
+        Arguments.of(
+            "SECOND",
+            new ExprDatetimeValue("2019-12-31 23:59:59"),
+            new ExprDatetimeValue("2020-01-01 00:00:00"),
+            1),
+        Arguments.of(
+            "DAY",
+            new ExprDatetimeValue("2019-12-31 23:59:59"),
+            new ExprDatetimeValue("2020-01-01 00:00:00"),
+            0),
+        Arguments.of(
+            "DAY",
+            new ExprDatetimeValue("2019-12-31 00:00:00"),
+            new ExprDatetimeValue("2020-01-01 00:00:00"),
+            1)
+    );
+  }
+
+  private static FunctionExpression timestampdiffQuery(FunctionProperties functionProperties,
+                                                       String unit,
+                                                       ExprValue datetimeExpr1,
+                                                      ExprValue datetimeExpr2) {
```

Comment:
```suggestion
                                                       ExprValue datetimeExpr2) {
```

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/251/commits/ed97a24f7ab7d040513331c17586b72bebdf6eb1